File: src/netconsole/models/api/device_compatibility.py
from __future__ import annotations

from pydantic import BaseModel, Field


class DeviceCompatibilityProfileDTO(BaseModel):
    profile_id: str
    vendor: str
    device_role: str
    platform: str
    validation_level: str
    capabilities: dict[str, str] = Field(default_factory=dict)


class DeviceCompatibilitySummaryDTO(BaseModel):
    generated_at: str
    profile_count: int
    platforms: list[str]
    roles: list[str]
    validation_levels: list[str]
    statement: str
    disclaimer: str
    profiles: list[DeviceCompatibilityProfileDTO]

File: apps/desktop_renderer/src/types/deviceCompatibility.ts
export interface DeviceCompatibilityProfile {
  profile_id: string
  vendor: string
  device_role: string
  platform: string
  validation_level: string
  capabilities: Record<string, string>
}

export interface DeviceCompatibilitySummary {
  generated_at: string
  profile_count: number
  platforms: string[]
  roles: string[]
  validation_levels: string[]
  statement: string
  disclaimer: string
  profiles: DeviceCompatibilityProfile[]
}
